src/psyclone/psyir/backend/fortran.py

```diff
@@ -306,6 +306,68 @@ def precedence(fortran_operator):
     raise KeyError()
 
 
+def add_accessibility_to_unknown_declaration(symbol):
+    '''
+    Utility that manipulates the unknown Fortran declaration for the supplied
+    Symbol so as to ensure that it has the correct accessibility specifier.
+    (This is required because we capture an 'unknown' Fortran declaration as
+    is and this may or may not include accessibility information.)
+
+    :param symbol: the symbol for which the declaration is required.
+    :type symbol: :py:class:`psyclone.psyir.symbols.Symbol`
+
+    :returns: Fortran declaration of the supplied symbol with accessibility \
+        information included (public/private).
+    :rtype: str
+
+    :raises TypeError: if the supplied argument is not a Symbol of \
+        UnknownFortranType.
+    :raises NotImplementedError: if the original declaration does not use \
+        '::' to separate the entity name from its type.
+    :raises InternalError: if the declaration stored for the supplied symbol \
+        contains accessibility information which does not match the \
+        visibility of the supplied symbol.
+
+    '''
+    if not isinstance(symbol, Symbol):
+        raise TypeError("Expected a Symbol but got '{0}'".format(
+            type(symbol).__name__))
+
+    if not isinstance(symbol.datatype, UnknownFortranType):
+        raise TypeError("Expected a Symbol of UnknownFortranType but symbol "
+                        "'{0}' has type '{1}'".format(symbol.name,
+                                                      symbol.datatype))
+
+    first_line = symbol.datatype.declaration.split("\n")[0]
```

Comment:
Would a case like the following make this break?
```
real, dimension(5,10,15), &
    intent(inout) :: x
```
If its to contrived to care about it will be enough to change the error message below from "in its original declaration" to "in the first line of its original declaration".

Also this could be wrapped with a IndexError try/catch in case declaration is empty and we can't do "[0]".

Reply:
I too worried about that but the Fortran reader deals with line-continuation symbols so your example would be a single line here. I've added a comment to this effect.
Your concern about an empty declaration is justified though so I've added a check for this with associated test.